### PR TITLE
Update Developer Guide

### PR DESCRIPTION
**Pull request by Benjykoh**
Updated DG for assign command's implementation to handle assigning tasks by module.

Updated diagrams and puml files to reflect this change.

### CODE REVIEW

**Review by Harish-coding — CHANGES_REQUESTED** [review]
Almost LGTM! The changes to DG are good! 👍 

The AssignCommandSequenceDiagram-2.png could be updated to reflect the following:

- [ ] The `parseStudentId(studentId)` could be within the alt box.
- [ ] In 'user inputs studentId and task' box, the activation bar for `ParserUtil#parseTask(task)` could be deactivated after returning task object.
- [ ] In 'user inputs moduleCode and task' box, - `AssignCommandParser -> ParserUtil: paserModuleCode(moduleCode)` - could be added. Then `activate ParserUtil`.


Please let me know if these suggestions are appropriate for the diagram.

**Comment by Benjykoh** [comment]
> Almost LGTM! The changes to DG are good! 👍
> 
> The AssignCommandSequenceDiagram-2.png could be updated to reflect the following:
> 
> * [ ]  The `parseStudentId(studentId)` could be within the alt box.
> * [ ]  In 'user inputs studentId and task' box, the activation bar for `ParserUtil#parseTask(task)` could be deactivated after returning task object.
> * [ ]  In 'user inputs moduleCode and task' box, - `AssignCommandParser -> ParserUtil: paserModuleCode(moduleCode)` - could be added. Then `activate ParserUtil`.
> 
> Please let me know if these suggestions are appropriate for the diagram.

Thank you so much for catching the errors! I will go correct the diagram!

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -508,27 +508,57 @@ After which, a new `AssignCommand` object will be created, and is subsequently e
...
+15. If a `Student` object with matching `studentId` or `moduleCode` is found the method uses `Person#isTaskAlreadyPresent(Task task)` 
+    method to check if the `task` is assigned.
     If no similar `task` is found, the following step will take place.
+
+   <div markdown="span" class="alert alert-info">:information_source:
+   <b>Note:</b>
+   If all the student(s) has already been assigned that task, then `DuplicateTaskException()` will be thrown. 
+   If some, not all students in the `moduleCode` has already been assigned that task, then `PartialDuplicateTaskException()` will be thrown.
+   </div>
+
```
Comment (medajet) [inline docs/DeveloperGuide.md:561]: I couldn't comment on the diagram itself, but regarding AssignCommandSequenceDiagram-1.png:

In the alt box, the condition text is overlapping with the activation bar, which might come across as a DG bug. I've faced a similar issue myself but it doesn't seem like there's a way to shift the text. A possible solution is to perhaps use another ref diagram? Just something for your consideration, though I'm sure there's probably a better solution out there.

**Review by medajet — APPROVED** [review]
DG changes look fine! In addition to what @Harish-coding has mentioned, I just have a small additional comment, but otherwise LGTM and approved 👍

**Pull request merged**